lang_python/parsing/Parser_python.mly

```diff
@@ -232,8 +232,6 @@ nl_or_stmt:
  | stmt    { $1 }
 
 sgrep_spatch_pattern:
- | test       EOF            { Expr $1 }
-
  | small_stmt EOF            { match $1 with [x] -> Stmt x | xs -> Stmts xs }
```

Comment:
I'm a bit worried about this one.
Might be good to do something like
match $1 with [ExprStmt e] -> Expr e | [x] -> Stmt x | xs -> Stmts xs 

Not sure why this did not introduce any regression, maybe because we do some kind of equivalence later and treat ExprStmt specially and run the deep expr matching by default for that, but I'd rather not rely here on that as this may change.

Reply:
Actually, did you try this with semgrep? Did you push this branch and tried if it introduces any regression in semgrep?

Reply:
I just made sure that pfff emitted the identical AST.

I think that YACC was always just eliminating this rule. Any single-expression sgrep pattern gets emitted as
```
S ( ExprStmt ( .... ) )
```
both before and after this change.

Reply:
I'll spend some points this morning validating that sgrep has no regressions just for extra safety though.

Reply:
Looks like the output of `make test` is unchanged before and after this diff.

Although there are currently two failing equivalence tests on `develop` that crop up for me.